src/display/ui.rs

```diff
@@ -182,4 +182,7 @@ where
     pub fn end(&mut self) {
         self.terminal.show_cursor().unwrap();
     }
+    pub fn update_interface_name(&mut self, interface_name: Option<String>) {
+        self.state.interface_name = interface_name;
+    }
```

Comment:
This kind of trivial setter method is very "Java-esque" and creates a lot of (frankly unnecessary) boilerplate code. We typically don't do this in Rust.